widget/iframe/src/main.ts

```diff
@@ -50,25 +135,58 @@ export class RangoWidget {
   /**
    * Creating an iframe node with desired attributes to load the widget.
    */
-  private createWidget(configuration?: WidgetConfig): HTMLIFrameElement {
+  private createWidget(configuration?: WidgetConfig) {
     if (!configuration) {
       throw new Error('Make sure you are passing configurations.');
     }
 
     const configs = this.encode(configuration);
-    const url = `${WIDGET_URL}?config=${configs}`;
-
+    const baseUrl = this.options?.widgetUrl || WIDGET_URL;
+    let url = `${baseUrl}?config=${configs}`;
     const widget = document.createElement('iframe');
     widget.setAttribute('id', RANGO_WIDGET_IFRAME_ID);
+
+    if (this.options?.clientUrl) {
+      url += `&clientUrl=${encodeURI(this.options.clientUrl)}`;
+      applyDynamicHeightStyles(widget);
+    } else {
+      applyDefaultStyles(widget);
+    }
     widget.src = url;
-    widget.style.backgroundColor = 'transparent';
-    widget.style.width = commonStyles.width;
-    widget.style.height = commonStyles.height;
-    widget.style.maxWidth = commonStyles.maxWidth;
-    widget.style.maxHeight = commonStyles.maxHeight;
-    widget.style.overflow = commonStyles.overflow;
-    widget.style.border = 'none';
-
-    return widget;
+    this.widget = widget;
+  }
+
+  /**
+   * Listening to messages and react to events sent by widget.
+   */
+  private listenToMessages() {
+    window.addEventListener('message', (e) => {
+      if (this.features.dynamicHeight && e.data.type === 'widget_height') {
+        this.onWidgetHeightChange(e.data);
+      }
+    });
+  }
+
+  private onWidgetHeightChange(message: any) {
```

Comment:
Is it possible to eliminate any type from this section?

Reply:
For avoiding duplicating types, I used any here. it should be export from `embeded`, but needs some extra works to export a new types of internal things. 
I tried to make it more clearer and write about next steps on top of if.